fabric/src/main/java/com/agricraft/agricraft/fabric/AgriCraftFabric.java

```
@@ -70,10 +70,10 @@ public void onInitialize() {
 		FabricLoader.getInstance().getModContainer("agricraft").ifPresent(agricraft -> {
 			for (ModContainer mod : FabricLoader.getInstance().getAllMods()) {
 				String modid = mod.getMetadata().getId();
-				if (!modid.equals("minecraft") && !modid.equals("agricraft")) {
+				if (!modid.equals("agricraft") && !modid.equals("minecraft") && agricraft.findPath("datapacks/"+modid).isPresent() && agricraft.findPath("resourcepacks/"+modid).isPresent()) {
```

`agricraft.findPath(“datapacks/”+modid).isPresent() && agricraft.findPath(“resourcepacks/”+modid).isPresent() ` The conditions should be on the following registration statement and not here

---

In place of `CoreConfig.enablePacksByDefault`? That would defeat the point of that config option...

---

registerBuiltinResourcePack

---

Like so?

```java
if (!modid.equals("agricraft") && !modid.equals("minecraft")) {
	if (agricraft.findPath("datapacks/"+modid).isPresent()) {
		ResourceManagerHelper.registerBuiltinResourcePack(new ResourceLocation("builtin", "agricraft_datapacks_" + modid), "datapacks/" + modid, agricraft, CoreConfig.enablePacksByDefault);
	}
	if (agricraft.findPath("resourcepacks/"+modid).isPresent()) {
		ResourceManagerHelper.registerBuiltinResourcePack(new ResourceLocation("builtin", "agricraft_resourcepacks_" + modid), "resourcepacks/" + modid, agricraft, CoreConfig.enablePacksByDefault);
	}
}
```

The thing is, all (current) AgriCraft datapacks require a corresponding resourcepack, so I figured it would be OK to fail hard if one or the other wasn't present.

---

Ya. Like so. Will work in more situation anyway.

---

I guess that does make it match the implementation on Forge...